File: code/networking/determine_endianess.c
#include <stdio.h>

int main(int argc, char const *argv[])
{
	unsigned int i = 1;
	char *x = (char*)&i;

	if(*x)
	{
		printf("Little Endian\n");
	}
	else
	{
		printf("Big Endian\n");
	}

	return 0;
}

File: code/networking/src/determine_endianess/determine_endianess.js
let endian = () => {
    let uInt32 = new Uint32Array([0x33883366]); //creating a 32 bit array and initializing it with arbitrary value
    let uInt8 = new Uint8Array(uInt32.buffer); //mapping the 32 bit array to 8 bit array
 
    if(uInt8[0] === 0x66) { //reading the first 4 bytes and determining endianness
        return 'Little Endian';
    } else {
        return 'Big Endian';
    }
};
 
endian();
